src/home/celebration/NftReward.test.tsx

```diff
@@ -0,0 +1,280 @@
+      <Provider store={createMockStore(mockStoreCelebrationReady)}>
+        <NftReward />
+      </Provider>
+    )
+
+    expect(queryByText('nftCelebration.rewardBottomSheet.title')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.description')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.cta')).toBeNull()
+  })
+
+  it('does not render when celebrated contract does not match with user nft', () => {
+    const { queryByText } = render(
+      <Provider store={createMockStore(mockStoreRewardReayWithDifferentNft)}>
+        <NftReward />
+      </Provider>
+    )
+
+    expect(queryByText('nftCelebration.rewardBottomSheet.title')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.description')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.cta')).toBeNull()
+  })
+
+  it('does not render when feature gate is closed', () => {
+    jest.mocked(getFeatureGate).mockReturnValue(false)
+
+    const { queryByText } = render(
+      <Provider store={createMockStore(mockStoreRewardReady)}>
+        <NftReward />
+      </Provider>
+    )
+
+    expect(queryByText('nftCelebration.rewardBottomSheet.title')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.description')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.cta')).toBeNull()
+  })
+
+  it('does not render if reward is already displayed', () => {
+    const { queryByText } = render(
+      <Provider store={createMockStore(mockStoreRewardDisplayed)}>
+        <NftReward />
+      </Provider>
+    )
+
+    expect(queryByText('nftCelebration.rewardBottomSheet.title')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.description')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.cta')).toBeNull()
+  })
+
+  it('does not render if reminder is already displayed', () => {
+    const { queryByText } = render(
+      <Provider store={createMockStore(mockStoreReminderDisplayed)}>
+        <NftReward />
+      </Provider>
+    )
+
+    expect(queryByText('nftCelebration.rewardBottomSheet.title')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.description')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.cta')).toBeNull()
+  })
+
+  it('does not render if expired', () => {
+    jest.useFakeTimers().setSystemTime(new Date('3001-01-01T00:00:00.000Z').getTime())
+
+    const { queryByText } = render(
+      <Provider store={createMockStore(mockStoreRewardReady)}>
+        <NftReward />
+      </Provider>
+    )
+
+    expect(queryByText('nftCelebration.rewardBottomSheet.title')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.description')).toBeNull()
+    expect(queryByText('nftCelebration.rewardBottomSheet.cta')).toBeNull()
+  })
+
+  it('hanldes the cta correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('3000-11-01T00:00:00.000Z').getTime())
```

Comment:
i think we need to reset this mock in the beforeEach's probably? otherwise this will affect the next tests, not a big deal now but could potentially cost some head scratching time in the future (like https://github.com/valora-inc/wallet/pull/4953#pullrequestreview-1895509126)